From ed888a1e885eda9e8d8313e6bd09c521f193f29d Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Sat, 5 Apr 2014 21:44:38 +0300
Subject: [PATCH] proper fix build due to divide u64/u32

Fix submitted by Jon Nettleton.
In some toolchains building might complain that _udivdi3 is not available. This fixes the issue

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 arch/arm/cpu/armv7/mx6/clock.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm/cpu/armv7/mx6/clock.c b/arch/arm/cpu/armv7/mx6/clock.c
index 734c054..14767df 100644
--- a/arch/arm/cpu/armv7/mx6/clock.c
+++ b/arch/arm/cpu/armv7/mx6/clock.c
@@ -5,6 +5,7 @@
  */
 
 #include <common.h>
+#include <div64.h>
 #include <asm/io.h>
 #include <asm/errno.h>
 #include <asm/arch/imx-regs.h>
@@ -123,8 +124,8 @@ static u32 mxc_get_pll_pfd(enum pll_clocks pll, int pfd_num)
 		return 0;
 	}
 
-	return (freq * 18) / ((div & ANATOP_PFD_FRAC_MASK(pfd_num)) >>
-			      ANATOP_PFD_FRAC_SHIFT(pfd_num));
+	return lldiv((freq * 18), ((div & ANATOP_PFD_FRAC_MASK(pfd_num)) >>
+			      ANATOP_PFD_FRAC_SHIFT(pfd_num)));
 }
 
 static u32 get_mcu_main_clk(void)
-- 
1.9.1

